### PR TITLE
IGAPP-796: Improve test performance

### PR DESCRIPTION
**Pull request by Taggotty**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

for me the jest update saves up to 20 seconds on native and web. It's still incredibly slow, taking around 23s to startup and even start running the first test. Partially it is because of the typescript transformation (which was apparently even disabled in native). I'm still working on further improvements on another branch. But split it as it would become quite big.

### CODE REVIEW

**Review by SteffiStuffel — APPROVED** [review]

File: native/jest.setup.ts [hunk 1]
```diff
@@ -2,7 +2,6 @@ import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/asy
 import '@testing-library/jest-native/extend-expect'
 import fs from 'fs'
 import path from 'path'
-
```
Comment (klinzo) [inline native/jest.setup.ts:5]: Revert, should be enforced by prettier anyway

File: native/src/routes/__tests__/Landing.spec.tsx [hunk 2]
```diff
@@ -12,10 +11,12 @@ import wrapWithTheme from '../../testing/wrapWithTheme'
...
 import Landing from '../Landing'
 
+const mockTheme = buildConfig().lightTheme
+
 jest.mock('react-i18next')
 jest.mock('styled-components', () => ({
   ...jest.requireActual('styled-components'),
-  useTheme: () => buildConfig().lightTheme
+  useTheme: () => mockTheme
 }))
```
Comment (klinzo) [inline native/src/routes/__tests__/Landing.spec.tsx:20]: Perhaps we could create a static mock for `styled-components`, lots of duplicated code. Oc no need to do this here.

**Review by klinzo — APPROVED** [review]
Nice work :)

**Review by f1sh1918 — APPROVED** [review]

File: native/src/routes/__tests__/Landing.spec.tsx (hunk 2 above)
Comment (klinzo) [reply native/src/routes/__tests__/Landing.spec.tsx:20]: https://issues.tuerantuer.org/browse/IGAPP-896

File: web/package.json [hunk 3]
```diff
@@ -106,9 +105,8 @@
     "html-loader": "^3.0.1",
     "html-webpack-plugin": "^5.5.0",
     "img-loader": "^4.0.0",
-    "jest": "^27.4.7",
     "jest-fetch-mock": "^3.0.3",
-    "jest-mock":"^27.4.6",
+    "jest-mock": "^27.4.6",
```
Comment (klinzo) [inline web/package.json:109]: Could be moved to root package.json as well but fine as it currently is

**Pull request merged**